npm-scripts/auto-pr-util.test.ts

```diff
@@ -0,0 +1,50 @@
+import { getRawIdFromTicketId, getTicketId } from "./auto-pr-util";
+
+describe("Auto PR Script Tests", () => {
+  describe('getTicketId', () => {
+    it('returns the provided ticket ID without searching the branch name', async () => {
+      const currentBranch = 'feature/TASK_999-implement-x';
+      const ticketId = 'TASK_123';
+      expect(getTicketId(currentBranch, ticketId)).toBe(ticketId);
+    });
+    
+    it('extracts and returns the ticket ID from the branch name when invalid ticket ID is provided', async () => {
+      const currentBranch = 'feature/TASK_456-implement-y';
+  
+      expect(getTicketId(currentBranch, "invalidticketid")).toBe('TASK_456');
+    });
+  
+    it('extracts and returns the ticket ID from the branch name when no ticket ID is provided', async () => {
+      const currentBranch = 'feature/TASK_456-implement-y';
+  
+      expect(getTicketId(currentBranch)).toBe('TASK_456');
+    });
+  
+    it('returns undefined and logs appropriate message when no ticket ID is found and none is provided', async () => {
+      const currentBranch = 'feature/implement-z';
+      expect(getTicketId(currentBranch)).toBeUndefined();
+    });
```

Comment:
Minor Nit: This doesn't seem to test that a message was logged directly. There are 2 ways you can achieve this:
1. Override `console.log` and assert it was called a certain way.
2. Use the logging library in this repo, add a test `LogHandler` that's a jest mock, and assert the mock was called with a specific log message.